packages/amplify-graphql-transformer-core/src/transformer-context/resolver.ts

```diff
@@ -504,17 +503,72 @@ export class TransformerResolver implements TransformerResolverProvider {
     }
   }
 
-  private stashStatementGenerator(
-    runtime?: CfnFunctionConfiguration.AppSyncRuntimeProperty,
-  ): (name: string, value?: string, object?: string) => string {
-    const jsStashStatement = (name: string, value?: string, object?: string): string => {
+  /**
+   * Generates a function to create stash statements based on the runtime.
+   *
+   * @param {CfnFunctionConfiguration.AppSyncRuntimeProperty} runtime - The AppSync runtime configuration.
+   * @returns {StashStatementGenerator} An object with methods to generate stash statements.
+   */
+  private generateStashStatementGenerator(runtime?: CfnFunctionConfiguration.AppSyncRuntimeProperty): StashStatementGenerator {
```

Comment:
Do any of our use cases accept double-quoted strings? If so, we'll need to escape the incoming values before passing them to the actual stasher (For JS, that'd be something like `value.replace(/"/g, '\\"')`; for VTL, it'd be something like `value.replace(/"/g, "${esc.q}")`, but we need to make sure that renders correctly in the resolver!

Reply:
Nothing within the `TransformerResolver` has double quoted strings. As you mentioned, we may have needs for that capability elsewhere. That's one of the reasons I opted to keep `generateStashStatementGenerator` private to the class, so as not to give the impression that this is an all-powerful / generic utility that can be used anywhere.

Do you see any concerns for the scope of stash statements generated within `TransformerResolver`? If not, I'd prefer to keep this as is for now and tackle generic handling as needed / separately.